openedx/core/djangoapps/credit/api/provider.py

```diff
@@ -30,66 +30,43 @@
+    **Use Cases**
 
-    Example:
-    >>> get_credit_providers()
-    [
-        {
-            "id": "hogwarts",
-            "display_name": "Hogwarts School of Witchcraft and Wizardry"
-        },
-        ...
-    ]
-
-    Returns: list
-    """
-    return CreditProvider.get_credit_providers()
+        Retrieve all available credit providers or filter on given providers_list.
 
+    **Arguments**:
 
-def get_credit_provider_info(provider_id):
-    """Retrieve the 'CreditProvider' model data against provided
-     credit provider.
+        providers_list (list of strings or None): contains list of ids of credit providers
+        or None.
 
-    Args:
-        provider_id (str): The identifier for the credit provider
+    **Returns**:
+        list of credit providers represented as dictionaries
 
-    Returns: 'CreditProvider' data dictionary
+    **Response Values**
 
-    Example Usage:
-        >>> get_credit_provider_info("hogwarts")
-        {
-            "provider_id": "hogwarts",
-            "display_name": "Hogwarts School of Witchcraft and Wizardry",
-            "provider_url": "https://credit.example.com/",
-            "provider_status_url": "https://credit.example.com/status/",
-            "provider_description: "A new model for the Witchcraft and Wizardry School System.",
-            "enable_integration": False,
-            "fulfillment_instructions": "
+        >>> get_credit_providers(['hogwarts'])
+        [
+            {
+                "id": "hogwarts",
+                "name": "Hogwarts School of Witchcraft and Wizardry",
+                "url": "https://credit.example.com/",
+                "status_url": "https://credit.example.com/status/",
+                "description: "A new model for the Witchcraft and Wizardry School System.",
+                "enable_integration": false,
+                "fulfillment_instructions": "
                 <p>In order to fulfill credit, Hogwarts School of Witchcraft and Wizardry requires learners to:</p>
                 <ul>
                 <li>Sample instruction abc</li>
                 <li>Sample instruction xyz</li>
                 </ul>",
-        }
-
+            },
+            ...
+        ]
     """
-    credit_provider = CreditProvider.get_credit_provider(provider_id=provider_id)
-    credit_provider_data = {}
-    if credit_provider:
-        credit_provider_data = {
-            "provider_id": credit_provider.provider_id,
-            "display_name": credit_provider.display_name,
-            "provider_url": credit_provider.provider_url,
-            "provider_status_url": credit_provider.provider_status_url,
-            "provider_description": credit_provider.provider_description,
-            "enable_integration": credit_provider.enable_integration,
-            "fulfillment_instructions": credit_provider.fulfillment_instructions
-        }
 
-    return credit_provider_data
+    return CreditProvider.get_credit_providers(providers_list)
```

Comment:
Just to be explicit that this is a keyword argument, please include the kwarg name:
```
CreditProvider.get_credit_providers(providers_list=providers_list)
```